src/PHPHtmlParser/Dom/HtmlNode.php

```diff
@@ -186,6 +186,9 @@ protected function clear()
         $this->innerHtml = null;
         $this->outerHtml = null;
         $this->text      = null;
+        if ( ! is_null($this->parent)) {
+            $this->parent->clear();
```

Comment:
Ignore me; I stand corrected.

This really is the best way to do it. It just looked weirder initially than it actually is.